docker-images/kafka/scripts/kafka_connect_run.sh

```diff
@@ -45,6 +45,29 @@ if [ "$KAFKA_CONNECT_METRICS_ENABLED" = "true" ]; then
     export KAFKA_OPTS
 fi
 
+if [ "$KAFKA_CONNECT_JMX_ENABLED" = "true" ]; then
+  KAFKA_JMX_OPTS="-Dcom.sun.management.jmxremote.port=9999 -Dcom.sun.management.jmxremote.rmi.port=9999 -Dcom.sun.management.jmxremote=true -Djava.rmi.server.hostname=$(hostname -i) -Djava.net.preferIPv4Stack=true"
+
+  if [ -n "$KAFKA_CONNECT_JMX_USERNAME" ]; then
+    # Secure JMX port on 9999 with username and password
+    JMX_ACCESS_FILE="/tmp/access.file"
+    JMX_PASSWORD_FILE="/tmp/password.file"
+
+    cat << EOF > "${JMX_ACCESS_FILE}"
+${KAFKA_CONNECT_JMX_USERNAME} readonly
+EOF
+
+    cat << EOF > "${JMX_PASSWORD_FILE}"
+$KAFKA_CONNECT_JMX_USERNAME $KAFKA_CONNECT_JMX_PASSWORD
+EOF
+    chmod 400 "${JMX_PASSWORD_FILE}"
+    KAFKA_JMX_OPTS="${KAFKA_JMX_OPTS} -Dcom.sun.management.jmxremote.access.file=${JMX_ACCESS_FILE} -Dcom.sun.management.jmxremote.password.file=${JMX_PASSWORD_FILE}  -Dcom.sun.management.jmxremote.authenticate=true"
+  else
+    # expose the port insecurely
+    KAFKA_JMX_OPTS="${KAFKA_JMX_OPTS} -Dcom.sun.management.jmxremote.authenticate=false"
+  fi
+fi
```

Comment:
This is almost exactly the same code as we have in `kafka_run.sh` (the only difference I can see is in the `KAFKA_CONNECT_JMX_ENABLED` vs `KAFKA_JMX_ENABLED`). I think it would be good to factor out some common script that we could `source` from both, rather than duplicating.

Reply:
hello @tombentley , what do you suggest as name for that script ? bests

Reply:
`build_jmx_opts.sh`?